corehq/apps/enterprise/views.py

```diff
@@ -323,3 +329,91 @@ def refresh_item(self, item_id):
 
     def post(self, *args, **kwargs):
         return self.paginate_crud_response
+
+
+@require_can_edit_or_view_web_users
+@require_superuser
+def enterprise_permissions(request, domain):
+    config = EnterprisePermissions.get_by_domain(domain)
+    all_domains = set(config.account.get_domains())
+    ignored_domains = all_domains - set(config.domains) - {config.source_domain}
+
+    context = {
+        'domain': domain,
+        'all_domains': sorted(all_domains),
+        'is_enabled': config.is_enabled,
+        'source_domain': config.source_domain,
+        'ignored_domains': sorted(list(ignored_domains)),
+        'controlled_domains': sorted(config.domains),
+        'current_page': {
+            'page_name': _('Enterprise Permissions'),
+            'title': _('Enterprise Permissions'),
+        }
+    }
+    return render(request, "enterprise/enterprise_permissions.html", context)
+
+
+@require_superuser
+@require_POST
+def disable_enterprise_permissions(request, domain):
+    config = EnterprisePermissions.get_by_domain(domain)
+    config.is_enabled = False
+    config.source_domain = None
+    config.save()
+
+    redirect = reverse("enterprise_permissions", args=[domain])
+    messages.success(request, _('Enterprise permissions have been disabled.'))
+    return HttpResponseRedirect(redirect)
+
+
+@require_superuser
+@require_POST
+def add_enterprise_permissions_domain(request, domain, target_domain):
+    config = EnterprisePermissions.get_by_domain(domain)
+
+    redirect = reverse("enterprise_permissions", args=[domain])
+    if target_domain not in config.account.get_domains():
+        messages.error(request, _("Could not add {}}.").format(target_domain))
+        return HttpResponseRedirect(redirect)
+
+    config.domains.append(target_domain)
```

Comment:
should this make an attempt to prevent adding duplicates?

Reply:
yeah, might as well - 11aa0a71edc